### ISSUE TITLE
kubevirt.io/client-go/kubecli is not compatible with k8s.io/client-go v0.21.0

### ISSUE THREAD

**Issue by erkanerol** [kubevirt/client-go#16]
There are some changes in apis with v0.21.0 and kubevirt.io/client-go is not compatible with those changes.

```
kubevirt.io/client-go/kubecli imports
        k8s.io/client-go/kubernetes/typed/batch/v2alpha1: module k8s.io/client-go@latest found (v11.0.0+incompatible, replaced by k8s.io/client-go@v0.21.0), but does not contain package k8s.io/client-go/kubernetes/typed/batch/v2alpha1
kubevirt.io/client-go/kubecli imports
        k8s.io/client-go/kubernetes/typed/discovery/v1alpha1: module k8s.io/client-go@latest found (v11.0.0+incompatible, replaced by k8s.io/client-go@v0.21.0), but does not contain package k8s.io/client-go/kubernetes/typed/discovery/v1alpha1
```

Some projects under kubevirt started using v0.21.0 (See https://github.com/kubevirt/hyperconverged-cluster-operator/blob/main/go.mod#L88) and this client should be updated as well.

**Comment by AlexsJones** [comment]
I would like to add that this issue is blocking further adoption of the client for us as we need to run other go.mod modules at v0.21.0

**Comment by mfranczy** [comment]
@fabiand any plans to fix the issue? :)

**Comment by rmohr** [comment]
@mfranczy feel free to bump the dependencies in kubevirt/kubevirt. But note that there is now a https://github.com/kubevirt/api repo with instructions on how to use the api only and how to generate a client for it.

**Comment by rmohr** [comment]
Oh and if the dependencies are still not trimmed down enough, consider replacing the cdi dependency with the latest one from https://github.com/kubevirt/containerized-data-importer-api which removes more k8s dependencies (https://github.com/kubevirt/containerized-data-importer/pull/2032)

**Comment by mfranczy** [comment]
@rmohr understood thanks, either me or @hdurand0710 will take care

**Comment by rmohr** [comment]
> This seems to be fixed by this commit : [10a1136](https://github.com/kubevirt/client-go/commit/10a1136aff3d82ec23276fcd2a0db43286fe5a3e)
> Thanks @rmohr

Great to hear. Still consider using our new api repo instead (https://github.com/kubevirt/api) :)
It is a more future prove way of interacting with different kubevirt and k8s versions.

/close
